Task 2/2.1/2.1.2/User.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace _2._1._2
+{
+    class User
+    {
+        public string name { get; }
+        private List<Figure> figures = new List<Figure>();
+
+        public User(string name)
+        {
+            this.name = name;
+        }
+
+        public void AddFigure(Figure figure)
+        {
+            figures.Add(figure);
+        }
```

Comment:
To me user's job is just to hold data that's related to user. Working with figures is another job that can be handled by another class (SRP Solid)
You could delegate it to AddHelper by passing figures collection to it. Or to DrawingArea, which could modify collection of current user